### ISSUE TITLE
plugins(wordpress): build fails when talking to the REST API because of a BOM character in the response

### ISSUE THREAD

**Issue by mewajda** [gatsbyjs/gatsby#21999]
## Description

When building or running develop the wordpress source plugin fails talking to the wordpress REST API because there is a BOM character in the first position that gets returned in the response.  Axios silently fails and response.data is a string instead of an object.  This causes issues for the rest of the processing that needs to happen.

You can see how I fixed the issue locally here: https://github.com/mewajda/gatsby/commit/c7acd101c0079ab2daff2451bf98b21d54b4a262

### Steps to reproduce

1. Run `gatsby new test-project`
2. Run `npm i gatsby-source-wordpress`
3. Add the following configuration:
  ```{
      resolve: `gatsby-source-wordpress`,
      options: {
        // your WordPress source
        baseUrl: `<somedomain.com>`,
        protocol: `http`,
        // is it hosted on wordpress.com, or self-hosted?
        hostingWPCOM: false,
        // does your site use the Advanced Custom Fields Plugin?
        useACF: true,
        verboseOutput: true,
        includedRoutes: [
          "**/categories",
          "**/posts",
          "**/pages",
          "**/media",
          "**/tags",
          "**/taxonomies",
          "**/users",
        ]
      },
    },
```
4. Run `gatsby develop`

### Expected result

The build should be successful and the graphql API should contain the wordpress nodes and fields

### Actual result

The build fails with the following result:
```
=START PLUGIN=====================================

Site URL: http://wisbouncin.com
Site hosted on Wordpress.com: false
Using ACF: true
Auth: false
Verbose output: true

Mama Route URL: http://wisbouncin.com/wp-json

Detected ACF to REST namespace: acf/v3.

 ERROR #11321  PLUGIN

"gatsby-source-wordpress" threw an error while running the sourceNodes lifecycle:

Cannot convert undefined or null to object

```

### Environment

  System:
    OS: Linux 4.15 Ubuntu 18.04.3 LTS (Bionic Beaver)
    CPU: (8) x64 Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
    Shell: 4.4.20 - /bin/bash
  Binaries:
    Node: 10.16.3 - /usr/local/bin/node
    npm: 6.9.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.15+ - /usr/bin/python
  Browsers:
    Chrome: 78.0.3904.97
  npmPackages:
    gatsby: ^2.19.28 => 2.19.28 
    gatsby-image: ^2.2.42 => 2.2.42 
    gatsby-plugin-manifest: ^2.2.42 => 2.2.42 
    gatsby-plugin-offline: ^3.0.35 => 3.0.35 
    gatsby-plugin-react-helmet: ^3.1.22 => 3.1.22 
    gatsby-plugin-sharp: ^2.4.5 => 2.4.5 
    gatsby-source-filesystem: ^2.1.48 => 2.1.48 
    gatsby-source-wordpress: ^3.1.63 => 3.1.63 
    gatsby-transformer-sharp: ^2.3.16 => 2.3.16 
  npmGlobalPackages:
    gatsby-cli: 2.10.0

**Comment by vladar** [comment]
Thanks for creating this issue. This is an interesting topic! And probably it is a bug somewhere on the API side, not Gatsby. Because JSON specification [directly forbids BOM](https://tools.ietf.org/html/rfc7159#section-8.1):

```
Implementations MUST NOT add a byte order mark to the beginning of a JSON text. 
```

At the same time, it allows parsers to account for it (but doesn't require it):
```
In the interests of interoperability, implementations that parse JSON texts
MAY ignore the presence of a byte order mark rather than treating it as an error.
```

I think adding this check makes sense as an improvement for Gatsby but the first thing to do in such a situation is to investigate the server-side issue (as it may break with other JSON parsers as well).

The PR with such an improvement for Gatsby is welcome though! 💜

**Comment by mewajda** [comment]
Thank you for the information.  I will see what I can find out on the Wordpress side. I have actually run into this issue before with a non-Wordpress API, but that also used php.

**Comment by mewajda** [comment]
After some more investigation and setting up a new wordpress site I was able to figure out the problem of why the BOM character was being returned.  Turns out that my `wp-config.php` was encoded using UTF-8 with BOM.  After, encoding it without the BOM everything started working fine.  I have no experience with PHP, so I am not sure why PHP would be prepending the BOM to REST responses just because a file was encoded in it.  Maybe someone else has more insights into why.

@vladar I am not sure if the PR is still valid anymore since its a pretty easy fix, but it seems like if any of the wordpress files are encoded with BOM then this issue would arise.  I tried a couple other http request libraries and none of them seem to handle this scenario.
